src/seedu/addressbook/data/person/Address.java

```diff
@@ -21,36 +35,72 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(address.trim())) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.block = new Block(matcher.group("block"));
+        this.street = new Street(matcher.group("street"));
+        this.unit = new Unit(matcher.group("unit"));
+        this.postalCode = new PostalCode(matcher.group("postalCode"));
```

Comment:
Avoid unnecessary use of `this` with fields: Use the `this` keyword only when a field is shadowed by a method or constructor parameter.

Reply:
I got it now, thank you.